CMakeLists.txt

```diff
@@ -42,53 +63,626 @@ if(USGSCSM_EXTERNAL_DEPS)
+    if(M_LIB)
+      target_link_libraries(proj PRIVATE -lm)
+    endif()
+    find_library(DL_LIB dl)
+    if(DL_LIB)
+      target_link_libraries(proj PRIVATE -ldl)
+    endif()
+  endif()
+  if(Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
+    target_link_libraries(proj PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+  endif()
+
+  target_include_directories(proj PRIVATE ${SQLITE3_INCLUDE_DIR})
+  target_link_libraries(proj PRIVATE ${SQLITE3_LIBRARY})
+
+  if(NLOHMANN_JSON STREQUAL "external")
+    target_compile_definitions(proj PRIVATE EXTERNAL_NLOHMANN_JSON)
+    target_link_libraries(proj
+      PRIVATE $<BUILD_INTERFACE:nlohmann_json::nlohmann_json>)
+  endif()
+
+  if(BUILD_SHARED_LIBS)
+    if(MSVC)
+      target_compile_definitions(proj PRIVATE PROJ_MSVC_DLL_EXPORT=1)
+    endif()
+  else()
+    target_compile_definitions(proj PUBLIC PROJ_DLL=)
+  endif()
+
+  ##############################################
+  # Core configuration summary
+  ##############################################
+  print_variable(PROJ_CORE_TARGET_OUTPUT_NAME)
+  print_variable(BUILD_SHARED_LIBS)
+  print_variable(PROJ_LIBRARIES)
+
+  set(PROJ_TARGET proj)
+  set(PROJ_INCLUDE_DIR /PROJ)
+
   # Nlohmann JSON
-  # Use the Nlohmann JSON version embedded in ALE
+  set(JSON_BuildTests OFF CACHE INTERNAL "")
+  add_subdirectory(json)
 
   # ALE
-  set(ALE_USE_EXTERNAL_EIGEN OFF)
-  set(ALE_USE_EXTERNAL_JSON OFF)
-  set(ALE_BUILD_LOAD OFF)
-  set(ALE_BUILD_TESTS OFF)
-  set(ALE_BUILD_DOCS OFF)
-  add_subdirectory(ale)
-  set(ALE_TARGET ale)
-
   # Use Eigen included with ALE
+  add_library (eigen INTERFACE)
   add_library (Eigen3::Eigen ALIAS eigen)
+  target_include_directories (eigen INTERFACE
+    ${CMAKE_CURRENT_SOURCE_DIR}/ale/eigen)
   set(EIGEN3_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/ale/eigen)
-endif(USGSCSM_EXTERNAL_DEPS)
 
-# Proj
-find_package(PROJ REQUIRED CONFIG)
-set(PROJ_TARGET PROJ::proj)
-
-add_library(usgscsm SHARED
-            src/UsgsAstroPlugin.cpp
-            src/UsgsAstroPluginSupport.cpp
-            src/UsgsAstroFrameSensorModel.cpp
-            src/UsgsAstroPushFrameSensorModel.cpp
-            src/UsgsAstroLsSensorModel.cpp
-            src/UsgsAstroProjectedSensorModel.cpp
-            src/UsgsAstroSarSensorModel.cpp
-            src/Distortion.cpp
-            src/Utilities.cpp
-            src/EigenUtilities.cpp)
+  set(ALE_BUILD_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/ale/include/ale")
+
+  set(ALE_SOURCE_FILES ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/InterpUtils.cpp
+                       ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/Rotation.cpp
+                       ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/Orientations.cpp
+                       ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/States.cpp
+                       ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/Util.cpp
+                       ${CMAKE_CURRENT_SOURCE_DIR}/ale/src/Vectors.cpp)
+
+  # set(ALE_HEADER_FILES ${ALE_BUILD_INCLUDE_DIR}/Rotation.h
+  #                      ${ALE_BUILD_INCLUDE_DIR}/Util.h)
+  
```

Comment:
Why is this commented out?

Reply:
It was a test to see if we needed the headers. I don't think the `ALE_HEADERS_FILES` variable is being used anywhere

Reply:
I will remove the commented lines